capstone/src/main/java/com/google/step/data/RestaurantHeader.java

```diff
@@ -29,12 +31,18 @@
   private final String name;
   private final GeoPt location;
   private final List<String> cuisine;
+  private final double metricsScore;
 
-  public RestaurantHeader(long restaurantKey, String name, GeoPt location, List<String> cuisine) {
+  public RestaurantHeader(long restaurantKey, String name, GeoPt location, List<String> cuisine, Optional<Double> metricsScore) {
```

Comment:
What's the general purpose of including this here? Presumably, scores should only travel in one direction (from your calculation to the search indexes), and from there, it should only ever be used in search queries to help sort by relevance.

Is this mainly to address the "isStruggling" stuff? (in which case, this is fine).

Reply:
Yeah, that was the idea. I was under the impression that the score was going to have front-end implications the whole time, we just hadn't had any scores before.

Reply:
Ok that's fine. Chat with Josh about it, we had discussed a translation into something more meaningful for the frontend, but maybe that translation happens outside of the headers, which is fine too.

Reply:
Did you mean to mention Martin here?